samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -44,7 +44,7 @@
 public class SamzaContainerController {
   private static final Logger log = LoggerFactory.getLogger(SamzaContainerController.class);
 
-  private final ExecutorService executorService;
```

Comment:
Can a single `SamzaContainerController` manage lifecycle of multiple containers? Otherwise, we can take the processorId, and make the executorService immutable?